src/nvim/option.c

```diff
@@ -2455,15 +2455,9 @@ did_set_string_option (
   }
   /* 'helpfile' */
   else if (varp == &p_hf) {
-    /* May compute new values for $VIM and $VIMRUNTIME */
-    if (didset_vim) {
-      vim_setenv("VIM", "");
-      didset_vim = FALSE;
-    }
-    if (didset_vimruntime) {
-      vim_setenv("VIMRUNTIME", "");
-      didset_vimruntime = FALSE;
-    }
+    // May compute new values for $VIM and $VIMRUNTIME
+    vim_setenv("VIM", "");
+    vim_setenv("VIMRUNTIME", "");
```

Comment:
With the old logic, if VIMRUNTIME was inherited from the environment, then `:set helpfile=...` will not unset it. Although this feature probably isn't used much, in general we want to avoid unsetting env vars because it affects `:term`, `:!`, etc. Especially we don't want to do that as a side-effect of `set helpfile`--at least not _more_ often...

It's great to remove globals, but these `didset_` globals are mostly harmless as long as they aren't used by many interacting components.